### PR TITLE
[9-11] project list and issue lists

### PR DESCRIPTION
**Pull request by Riwuko**
[9] Added project list with a possibility to assing users to a project
[11] Added issue list with a possibility to assing users to an issue

*Description:* This branch adds project and issue lists. User can see projects he owns or is assigned to. In project detail there is information about its users and issues. Single issue has also information about its assignee and the project it belongs to.

### CODE REVIEW

File: kanban/api/serializers.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+from django.shortcuts import get_object_or_404
```
Comment (nitz14) [inline kanban/api/serializers.py:1]: please use Isort :)

File: kanban/api/views/issue.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from kanban.api.serializers import IssueSerializer
+from kanban.models.issue import Issue
+from rest_framework import viewsets
+
+
+class IssueViewSet(viewsets.ModelViewSet):
+    queryset = Issue.objects.all()
+    serializer_class = IssueSerializer
```
Comment (nitz14) [inline kanban/api/views/issue.py:8]: Should this View will be open for all users ?

File: kanban/api/views/issue.py [hunk 3]
```diff
@@ -0,0 +1,8 @@
+from kanban.api.serializers import IssueSerializer
```
Comment (nitz14) [inline kanban/api/views/issue.py:1]: isort

File: kanban/api/views/project.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+from rest_framework import viewsets
+
+
+class ProjectViewSet(viewsets.ModelViewSet):
+    queryset = Project.objects.all()
+    serializer_class = ProjectSerializer
+
+    def get_queryset(self):
+        return Project.objects.filter(
+            Q(owner=self.request.user) | Q(users__pk=self.request.user.pk)
```
Comment (nitz14) [inline kanban/api/views/project.py:13]: maybe we can add owner automatically to users list ?

File: kanban/api/views/project.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from django.db.models import Q
+from kanban.api.serializers import ProjectSerializer
+from kanban.models.project import Project
```
Comment (nitz14) [inline kanban/api/views/project.py:3]: isort

File: kanban/migrations/0006_auto_20200929_1246.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.2.16 on 2020-09-29 10:46
```
Comment (nitz14) [inline kanban/migrations/0006_auto_20200929_1246.py]: Can we create 1 migrations ?

File: kanban/api/views/project.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+from rest_framework import viewsets
+
+
+class ProjectViewSet(viewsets.ModelViewSet):
+    queryset = Project.objects.all()
+    serializer_class = ProjectSerializer
```
Comment (nitz14) [inline kanban/api/views/project.py:9]: should this endpoint will open ?

File: kanban/api/views/issue.py (hunk 2 above)
Comment (Riwuko) [reply kanban/api/views/issue.py:8]: All views are by default from settings closed to unauthorized users:
 ("DEFAULT_PERMISSION_CLASSES": [
        "rest_framework.permissions.IsAuthenticated", )

File: kanban/api/serializers.py (hunk 1 above)
Comment (Riwuko) [reply kanban/api/serializers.py:1]: I am and that is how it sorts my imports. Should I change some settings or sth?

File: kanban/api/views/project.py (hunk 4 above)
Comment (Riwuko) [reply kanban/api/views/project.py:13]: But if owner will for example remove himself from user list then he won't be able to see project, and yet he will still be owner? Is that not strange situation if owner has no rights?

File: kanban/api/views/project.py (hunk 7 above)
Comment (nitz14) [reply kanban/api/views/project.py:9]: ok

File: kanban/api/serializers.py (hunk 1 above)
Comment (nitz14) [reply kanban/api/serializers.py:1]: sth is wrong this Isort :/

Comment (Riwuko) [reply kanban/api/serializers.py:1]: so what should I do? :(

File: config/urls.py [hunk 8]
```diff
@@ -20,4 +20,5 @@
     path("admin/", admin.site.urls),
     path("api/", include("user_account.api.urls", "user_account_api")),
     path("social/", include("social_django.urls", namespace="social")),
+    path("", include("kanban.api.urls", "kanban_api")),
```
Comment (robertmusil) [inline config/urls.py:23]: why not in `/api` path?

File: kanban/api/serializers.py [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+class UserAccountRelatedField(serializers.RelatedField):
+    def display_value(self, instance):
+        return instance
+
+    def to_representation(self, value):
+        return str(value)
+
+    def to_internal_value(self, data):
+        try:
+            user = UserAccount.objects.get(email=data)
```
Comment (robertmusil) [inline kanban/api/serializers.py:17]: How many queries are fired to fetch a single issue?

**Review by robertmusil — APPROVED** [review]

**Review by nitz14 — APPROVED** [review]

**Pull request merged**